### PR TITLE
spec: added missing hyperlink and some clean-up on ABCI spec

### PR DESCRIPTION
**Pull request by andynog**
close: #2679 

Added hyperlink and cleaned-up hidden comments

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments
- [ ] Title follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) spec

### CODE REVIEW

File: spec/abci/abci++_app_requirements.md [hunk 1]
```diff
@@ -276,20 +276,6 @@ time.
 
 The existence of this global mutex means Go application developers can get thread safety for application state by routing all reads and writes through the ABCI system. Thus it may be unsafe to expose application state directly to an RPC interface, and unless explicit measures are taken, all queries should be routed through the ABCI Query method.
 
-<!--
```
Comment (sergio-mena) [inline spec/abci/abci++_app_requirements.md:279]: For `v0.38.x` this is good.
However, for `main` and `v1.x` you may want to explain here the new unsync client (check [here)](https://github.com/cometbft/cometbft/blob/v1.0.0-alpha.2/abci/client/unsync_local_client.go#L32) we now have. You would need to adapt the commented text: in `v0.36.x` the former Tendermint Core team decided to remove the mutex altogether, whereas `v1.x` gives the flexibility of keeping the mutex or not, but providing the extra ABCI client.

File: spec/abci/abci++_app_requirements.md [hunk 2]
```diff
@@ -594,25 +580,35 @@ all full nodes have the same value at a given height.
...
+5. [EvidenceParams.MaxAgeNumBlocks](#evidenceparamsmaxagenumblocks)
+6. [EvidenceParams.MaxBytes](#evidenceparamsmaxbytes)
+7. [ValidatorParams.PubKeyTypes](#validatorparamspubkeytypes)
+8. [VersionParams.App](#versionparamsapp)
+
+##### ABCIParams.VoteExtensionsEnableHeight
+
+This parameter is either 0 or a positive height at which vote extensions
+become mandatory. If the value is zero (which is the default), vote
+extensions are not required. Otherwise, at all heights greater than the
```
Comment (sergio-mena) [inline spec/abci/abci++_app_requirements.md:598]:
```suggestion
extensions are not expected. Otherwise, at all heights greater than the
```

**Review by sergio-mena — APPROVED** [review]

File: spec/abci/abci++_app_requirements.md (hunk 1 above)
Comment (andynog) [reply spec/abci/abci++_app_requirements.md:279]: I can open a separate issue for this on v1, thanks for pointing that out

Comment (sergio-mena) [reply spec/abci/abci++_app_requirements.md:279]: Thanks

**Pull request merged**